sys/userlib/src/kipc.rs

```diff
@@ -24,6 +26,20 @@ pub fn read_task_status(task: usize) -> abi::TaskState {
     ssmarshal::deserialize(&response[..len]).unwrap_lite().0
 }
 
+pub fn find_faulted_task(task: usize) -> Option<NonZeroUsize> {
+    // Coerce `task` to a known size (Rust doesn't assume that usize == u32)
+    let task = task as u32;
+    let mut response = 0_u32;
+    let (_, _) = sys_send(
+        TaskId::KERNEL,
+        Kipcnum::ReadTaskStatus as u16,
```

Comment:
uhh...this looks like the wrong `Kipcnum` variant?
```suggestion
        Kipcnum::FindFaultedTask as u16,
```

Reply:
Good catch! I've mostly tested this with exhubris where I used the right number. The perils of a fork!